sentry_sdk/api.py

```diff
@@ -156,111 +208,145 @@ def push_scope(  # noqa: F811
     pass
 
 
-@hubmethod
 def push_scope(  # noqa: F811
     callback=None,  # type: Optional[Callable[[Scope], None]]
 ):
     # type: (...) -> Optional[ContextManager[Scope]]
-    return Hub.current.push_scope(callback)
+    """
+    Pushes a new layer on the scope stack.
+
+    :param callback: If provided, this method pushes a scope, calls
+        `callback`, and pops the scope again.
+
+    :returns: If no `callback` is provided, a context manager that should
+        be used to pop the scope again.
+    """
+
+    class _ScopeManager:
+        def __init__(self):
+            # type: () -> None
+            pass
+
+        def __enter__(self):
+            # type: () -> Scope
+            return Scope.get_isolation_scope()
+
+        def __exit__(self, exc_type, exc_value, tb):
+            # type: (Any, Any, Any) -> None
+            pass
```

Comment:
Why not like this?
```suggestion
    @contextmanager
    def scope_manager():
        yield Scope.get_isolation_scope()
```

We would need to ensure `contextlib.contextmanager` is imported, and add the appropriate type hints.

Reply:
Good idea. And if this would not be Hub based stuff I would change it, but because this code will be removed anyhow, I will not touch it.